src/app/public/modules/data-manager/data-manager.service.ts

```diff
@@ -0,0 +1,114 @@
+import {
+  Injectable
+} from '@angular/core';
+
+import {
+  BehaviorSubject
+} from 'rxjs/BehaviorSubject';
+
+import {
+  SkyDataManagerConfig,
+  SkyDataManagerState,
+  SkyDataManagerStateChange,
+  SkyDataViewConfig,
+  SkyDataViewState
+} from './models';
+
+import {
+  Observable
+} from 'rxjs/Observable';
+
+import {
+  filter,
+  map
+} from 'rxjs/operators';
+
+@Injectable()
+export class SkyDataManagerService {
+
+  private readonly activeViewId: BehaviorSubject<string> = new BehaviorSubject<string>(undefined);
+
+  private readonly dataManagerConfig = new BehaviorSubject<SkyDataManagerConfig>(undefined);
+
+  private readonly views = new BehaviorSubject<SkyDataViewConfig[]>([]);
+
+  private readonly dataStateChange =
+    new BehaviorSubject<SkyDataManagerStateChange>(new SkyDataManagerStateChange(new SkyDataManagerState({}), 'defaultState'));
+
+  public getCurrentDataState(): SkyDataManagerState {
```

Comment:
Are all of these methods intended to be used by consumers? I only ask because with the old list-builder, we unintentionally had a few (undocumented) public methods on our list component that consumers started using without our knowledge. I'd suggest giving these methods some nice jsdoc comments to get ahead of that this time around 😄 .

Reply:
they are! jsdoc comments to come